op_builder/builder.py

```diff
@@ -107,6 +107,7 @@ def __init__(self, name):
         self.name = name
         self.jit_mode = False
         self.build_for_cpu = False
+        self.enable_bf16 = False
```

Comment:
@jeffra This is being done in the constructor for OpBuilder class, even though the `enable_bf16` attributed is really only checked for `CUDAOpBuilder`: https://github.com/microsoft/DeepSpeed/blob/82991df05419828d65cbb658efc01b50c135ea00/op_builder/builder.py#L492

The issue is that the initialization of that attribute happens inside https://github.com/microsoft/DeepSpeed/blob/82991df05419828d65cbb658efc01b50c135ea00/op_builder/builder.py#LL517C9-L517C32, which is only called for CUDA.

Would you prefer to initialize this in OpBuilder or CUDAOpBuilder?

Reply:
cc @tjruwase